experimental/segmenter/src/provider.rs

```diff
@@ -243,3 +243,24 @@ impl ResourceProvider<SentenceBreakDataV1Marker> for RuleBreakDataProvider {
         })
     }
 }
+
+/// char16trie data for dictionary break
+#[icu_provider::data_struct(UCharDictionaryBreakDataV1Marker = "segmenter/char16trie@1")]
+pub struct UCharDictionaryBreakDataV1<'data> {
+    /// Dictionary data of char16trie.
+    #[cfg_attr(feature = "provider_serde", serde(borrow))]
+    pub trie_data: ZeroVec<'data, u16>,
+}
+
+// Test data of thai dictionary
+const THAI_DICTIONARY: &[u8; 224978] = include_bytes!("../tests/testdata/thai.dict");
```

Comment:
Suggestion move the `unsafe` up here, like

```suggestion
// Safe because all byte sequences are valid u16::ULE
const THAI_DICTIONARY: [u16::ULE; 112489] = unsafe {
    core::mem::transmute(*include_bytes!("../tests/testdata/thai.dict"))
};
```

(we will get rid of the unsafe in #1647)

Reply:
Thanks, I will use `[<u16 as AsULE>::ULE; ...]`.